Cython/Includes/cpython/time.pxd

```diff
@@ -30,15 +30,17 @@ cdef inline int _raise_from_errno() except -1 with gil:
     return <int> -1  # Let the C compiler know that this function always raises.
 
 
-cdef inline tm localtime() nogil except *:
+cdef inline tm localtime(time_t tic) nogil except *:
     """
     Analogue to the stdlib time.localtime.  The returned struct
     has some entries that the stdlib version does not: tm_gmtoff, tm_zone
     """
     cdef:
-        time_t tic = <time_t>time()
         tm* result
 
+    if tic is None:
```

Comment:
A C integer type cannot be None. It's a little tricky to make this argument optional, though, because there is not really a safe default value. Even `0` or `((time_t)-1)` may be reasonable values for users to ask for.

Reply:
i guess we can make it required and document that users can pass e.g. `int(time.time())` to duplicate the no-arg behavior?

Reply:
Or, given that this is a performance oriented replacement that users *chose* to use, we could just document that `((time)-1)` behaves differently and returns the current time.

Reply:
either way i guess.  i lean to towards the not-special-casing.

Reply:
Would a fused function work? I don't think Cython has an easy way to specify NoneType though, so it'd have to be `object` check for None otherwise convert to int.

I don't know if there's an issue with inline fused functions in pxds.

Reply:
I'm not completely sure that's a *good* idea, by the way. But it is an idea...

Reply:
The function is intended to *avoid* type conversions. That rules out Python object input.

I'd rather allow C function overloads here.

Reply:
the most recent comment took out the None check, so should always require time_t

Reply:
Requiring the argument is simple enough. I don't know which use case is more common: getting the current time or asking for a specific time. If getting the current time is very common, then I'd rather have the argument optional.

But, requiring it now still allows us to make it optional at some point, so we don't lose anything. It'd just be a little less convenient than it could be.

Reply:
> the most recent comment took out the None check, so should always require time_t

However, the test also shows that the work-around isn't as simple as that. `time()` returns a `double`, not a `time_t`, so users need at least an additional cast, or call `libc.time.time(NULL)` instead of `cpython.time.time()`. That makes it another bit less attractive to require the argument.